<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Carnivora_coocurrence - A repository for:</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Carnivora_coocurrence</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./C_modelling_2023.html">
 <span class="menu-text">Competitive Modelling</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./P_modelling_2023.html">
 <span class="menu-text">Predato-prey Modelling</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#a-repository-for" id="toc-a-repository-for" class="nav-link active" data-scroll-target="#a-repository-for">A repository for:</a>
  <ul class="collapse">
  <li><a href="#links-to-different-parts-of-the-page" id="toc-links-to-different-parts-of-the-page" class="nav-link" data-scroll-target="#links-to-different-parts-of-the-page">Links to different parts of the page</a></li>
  <li><a href="#description-of-the-repository" id="toc-description-of-the-repository" class="nav-link" data-scroll-target="#description-of-the-repository">Description of the repository</a>
  <ul class="collapse">
  <li><a href="#the-root-folder" id="toc-the-root-folder" class="nav-link" data-scroll-target="#the-root-folder">The root folder</a></li>
  <li><a href="#the-data-folder-.data" id="toc-the-data-folder-.data" class="nav-link" data-scroll-target="#the-data-folder-.data">The data folder (<code>./Data</code>)</a></li>
  <li><a href="#files-for-statistical-modeling" id="toc-files-for-statistical-modeling" class="nav-link" data-scroll-target="#files-for-statistical-modeling">Files for statistical modeling</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">A repository for:</h1>
<p class="subtitle lead">From co-occurrence patterns to agonistic interactions. A meta-analysis of carnivore mammals camera trap research</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Add Zenodo Doi</p>
<section id="a-repository-for" class="level1">
<h1>A repository for:</h1>
<p>From co-occurrence patterns to agonistic interactions. A meta-analysis of carnivore mammals camera trap research</p>
<section id="links-to-different-parts-of-the-page" class="level2">
<h2 class="anchored" data-anchor-id="links-to-different-parts-of-the-page">Links to different parts of the page</h2>
<ol type="1">
<li><a href="#description-of-the-repository">Description of the repository</a></li>
<li><a href="#the-root-folder">The root folder</a></li>
<li><a href="#the-data-folder-.data">The data folder (<code>./data</code>)</a></li>
<li><a href="#files-for-statistical-modeling">Files for statistical modeling</a></li>
</ol>
</section>
<section id="description-of-the-repository" class="level2">
<h2 class="anchored" data-anchor-id="description-of-the-repository">Description of the repository</h2>
<p>This repository contains the data and code used in <em>From co-occurrence patterns to agonistic interactions. A meta-analysis of carnivore mammals camera trap research</em>. The data come from a literature review of peer-reviewed articles focused on the study of spatial or temporal co-occurrence of mammals of the order Carnivora with camera traps. All data used in the modeling process are located in the <code>./Data/</code> folder. Inside the <code>./Data/Literature</code> folder you will find the information of the articles selected in the literature search. The <code>./Data/Model data</code> folder contains the files with the information used to perform the statistical modeling. The code is available online in html form for each interaction and in Quarto script format in the root folder.</p>
<p><a href="#a-repository-for">Back to table of contents ⤒</a></p>
<section id="the-root-folder" class="level3">
<h3 class="anchored" data-anchor-id="the-root-folder">The root folder</h3>
<p>The root folder contains the following files: <code>_quarto.yml</code>, <code>styles.css</code> and <code>.gitignore</code>, which are configuration files for rendering Quarto files (<code>*.qmd</code>) and GitHub configurations respectively. The Readme file (<code>README.md</code>) contains the same information of the <code>index.qmd</code> file, which is the description of the repository. The code is in Quarto format and contains statistical modeling procedures for competitive (<code>C_modelling_2023.qmd</code>) and predator-prey (<code>P_modelling_2023.qmd</code>) interactions. Finally the <code>Carnivora_coocurrence.Rproj</code> is used to open the project in Rstudio.</p>
<p>The scripts are in R language but require <a href="https://quarto.org/">Quarto</a> to be installed in Rstudio to run. However the corresponding <code>.htmls</code> files contain all the code information (including results) without the need to open R.</p>
<p><a href="#a-repository-for">Back to table of contents ⤒</a></p>
</section>
<section id="the-data-folder-.data" class="level3">
<h3 class="anchored" data-anchor-id="the-data-folder-.data">The data folder (<code>./Data</code>)</h3>
<p>This folder has 5 files and 2 sub-folders.</p>
<p><strong><code>./Data/Literature/bib_database.csv</code></strong>: Database in csv format with the 314 fully scanned articles. The base columns have the following information</p>
<table class="table">
<caption>Column description of bib_database.csv</caption>
<colgroup>
<col style="width: 24%">
<col style="width: 75%">
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>label</td>
<td>Identifier of each article</td>
</tr>
<tr class="even">
<td>title</td>
<td>Complete title</td>
</tr>
<tr class="odd">
<td>author</td>
<td>List of authors</td>
</tr>
<tr class="even">
<td>journal</td>
<td>publication journal</td>
</tr>
<tr class="odd">
<td>abstract</td>
<td>abstract of each article</td>
</tr>
<tr class="even">
<td>year</td>
<td>year of publication</td>
</tr>
<tr class="odd">
<td>doi</td>
<td>digital object indentification</td>
</tr>
<tr class="even">
<td>volume</td>
<td>journal volume</td>
</tr>
<tr class="odd">
<td>language</td>
<td>language of publication</td>
</tr>
<tr class="even">
<td>issue</td>
<td>journal issue</td>
</tr>
<tr class="odd">
<td>Selected</td>
<td>Selected or not selected during the screening process</td>
</tr>
</tbody>
</table>
<p><strong><code>./Data/Model Data/</code></strong> : This folder contains 4 files containing all the information extracted from the selected articles. <code>spatC_db.csv</code> and <code>tempC_db.csv</code> contain the information on the competitive interactions of carnivorous mammals in the spatial and temporal dimension respectively. While, <code>spatP_db.csv</code> and <code>tempP_db.csv</code> contain the information on predator-prey interactions in the spatial and temporal dimension respectively. The description of the variables of the four bases are described in the following table</p>
<table class="table">
<caption>Description of the columns present in the databases of the model data folder</caption>
<colgroup>
<col style="width: 7%">
<col style="width: 65%">
<col style="width: 19%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Description</th>
<th>Source</th>
<th>Interaction</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SIF</td>
<td>Species interaction factor derived from multi-species occupancy models; SIF&lt;1 species avoidance, SIF:1 random pattern, SIF&gt;1 Aggregation pattern</td>
<td>Each study or calculated</td>
<td>Both</td>
</tr>
<tr class="even">
<td>SIF_ic</td>
<td>SIF 95% confidence intervals</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>Ov_coeff</td>
<td>Overlapping delta coefficient: 0 indicating no overlapping in species activity patterns, 1 indicating total overlapping</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>Ov_ic</td>
<td>Overlapping delta coefficient 95% confidence intervals</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>D_competitor</td>
<td>Scientific name of the bigger or dominant competitor</td>
<td>Each study</td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>S_competitor</td>
<td>Scientific name of the subordinate or smaller competitor</td>
<td>Each study</td>
<td>Competitive</td>
</tr>
<tr class="odd">
<td>D_family</td>
<td>Family of the dominant competitor</td>
<td>Each study</td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>S_family</td>
<td>Family of the subordinate competitor</td>
<td>Each study</td>
<td>Competitive</td>
</tr>
<tr class="odd">
<td>D_genus</td>
<td>Genus of the dominant competitor</td>
<td>Each study</td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>S_genus</td>
<td>Genus of the subordinate competitor</td>
<td>Each study</td>
<td>Competitive</td>
</tr>
<tr class="odd">
<td>p_distance</td>
<td>Pairwise mitochondrial DNA phylogenetic distances</td>
<td><span class="citation" data-cites="hassanin2021">(<a href="#ref-hassanin2021" role="doc-biblioref">Hassanin et al. 2021</a>)</span></td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>mass_ratio</td>
<td><p>For competitive interactions=ln(Dominant competitor body mass (g)/ Subordinate competitor body mass (g)).</p>
<p>For predator-prey interactions= ln(Predator body mass (g) / Prey body mass (g))</p></td>
<td>Calculated</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>D_mass</td>
<td>Dominant competitor body mass (g)</td>
<td>COMBINE data base <span class="citation" data-cites="soria2021">(<a href="#ref-soria2021" role="doc-biblioref">Soria et al. 2021</a>)</span></td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>S_mass</td>
<td>Subordinate competitor body mass (g)</td>
<td>COMBINE data base <span class="citation" data-cites="soria2021">(<a href="#ref-soria2021" role="doc-biblioref">Soria et al. 2021</a>)</span></td>
<td>Competitive</td>
</tr>
<tr class="odd">
<td>D_diet</td>
<td>Dominant competitor diet category</td>
<td>PHYLACINE trait database <span class="citation" data-cites="faurby2020">(<a href="#ref-faurby2020" role="doc-biblioref">Faurby et al. 2020</a>)</span></td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>S_diet</td>
<td>Subordinate competitor diet category</td>
<td>PHYLACINE trait database <span class="citation" data-cites="faurby2020">(<a href="#ref-faurby2020" role="doc-biblioref">Faurby et al. 2020</a>)</span></td>
<td>Competitive</td>
</tr>
<tr class="odd">
<td>diet_dist</td>
<td>Dominant and subordinate competitor category diets: Same if pair of competitors shares de diet category, different if not</td>
<td>calculated</td>
<td>Competitive</td>
</tr>
<tr class="even">
<td>Pred_sp</td>
<td>Scientific name of the predator</td>
<td>Each study</td>
<td>Predator-prey</td>
</tr>
<tr class="odd">
<td>Prey_sp</td>
<td>Scientific name of the prey</td>
<td>Each study</td>
<td>Predator-prey</td>
</tr>
<tr class="even">
<td>Pred_family</td>
<td>Family of the predator</td>
<td>Each study</td>
<td>Predator-prey</td>
</tr>
<tr class="odd">
<td>Prey_family</td>
<td>Family of the prey</td>
<td>Each study</td>
<td>Predator-prey</td>
</tr>
<tr class="even">
<td>Pred_genus</td>
<td>Genus of the predator</td>
<td>Each study</td>
<td>Predator-prey</td>
</tr>
<tr class="odd">
<td>Prey_genus</td>
<td>Genus of the prey</td>
<td>Each study</td>
<td>Predator-prey</td>
</tr>
<tr class="even">
<td>Pred_mass</td>
<td>Predator body mass (g)</td>
<td>COMBINE data base <span class="citation" data-cites="soria2021">(<a href="#ref-soria2021" role="doc-biblioref">Soria et al. 2021</a>)</span></td>
<td>Predator-prey</td>
</tr>
<tr class="odd">
<td>P_hunt</td>
<td>Predator hunting strategy: pursuit, ambush, group hunting</td>
<td><span class="citation" data-cites="hirt2020">(<a href="#ref-hirt2020" role="doc-biblioref">Hirt et al. 2020</a>)</span> and other sources</td>
<td>Predator-prey</td>
</tr>
<tr class="even">
<td>Prey_mass</td>
<td>Prey body mass (g)</td>
<td>COMBINE data base <span class="citation" data-cites="soria2021">(<a href="#ref-soria2021" role="doc-biblioref">Soria et al. 2021</a>)</span></td>
<td>Predator-prey</td>
</tr>
<tr class="odd">
<td>Label</td>
<td>Article identifier</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>Lon</td>
<td>Longitude of the center of the locality or each study locality in decimal coordinates</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>Lat</td>
<td>Latitude of the center of the locality or each study locality in decimal coordinates</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>Continent</td>
<td>Continent in which the research is conducted</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>Country</td>
<td>Country in which the research is carried out</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>Locality</td>
<td>Location where the co-occurrence measurements were obtained.</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>Samp_dur</td>
<td>Sampling duration in months</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>N_stations</td>
<td>Number of sampling stations</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>Effort</td>
<td>Sampling effort calculated as number of sampling stations per number of camera days</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>Avg_dist</td>
<td>Average distance between sampling stations in km</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="odd">
<td>S_area</td>
<td>Sampling area in km^2</td>
<td>Each study</td>
<td>Both</td>
</tr>
<tr class="even">
<td>Pred_evidence</td>
<td>Direct evidence of predation</td>
<td>Global Interaction data base <span class="citation" data-cites="poelen2014">(<a href="#ref-poelen2014" role="doc-biblioref">Poelen, Simons, and Mungall 2014</a>)</span></td>
<td>Predator-prey</td>
</tr>
<tr class="odd">
<td>Hunt_sourve</td>
<td>Bibliographic source of the hunting strategy</td>
<td>Literature</td>
<td>Predator-prey</td>
</tr>
</tbody>
</table>
<p><a href="#a-repository-for">Back to table of contents ⤒</a></p>
</section>
<section id="files-for-statistical-modeling" class="level3">
<h3 class="anchored" data-anchor-id="files-for-statistical-modeling">Files for statistical modeling</h3>
<p>The <code>qmd</code> files contain the entire statistical modeling prthe respective results.</p>
<p><br>
<strong><code>C_modelling_2023.qmd</code></strong> : This file contain the modeling process performed for the co-occurrence data of competing species of carnivorous mammals. It includes both the modeling of the spatial co-occurrence data and the temporal co-occurrence data. The script requires the following data files: <code>Data/Model data/SpatC_db.csv</code> and <code>Data/Model data/tempC_db.csv</code>. Can be viewed without opening R using the <code>C_modelling_2023.html</code> file and the online page.<br>
<br>
<strong><code>P_modelling_2023.qmd</code></strong> : This file contain the modeling process performed for the co-occurrence data of Carnivora predators an their mammals preys. It includes both the modeling of the spatial co-occurrence data and the temporal co-occurrence data. The script requires the following data files: <code>Data/Model data/PpatC_db.csv</code> and <code>Data/Model data/tempP_db.csv</code>. Can be viewed without opening R using the <code>P_modelling_2023.html</code> file and the online page.</p>
<p><a href="#a-repository-for">Back to table of contents ⤒</a></p>



</section>
</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-faurby2020" class="csl-entry" role="doc-biblioentry">
Faurby, Søren, Rasmus Ø Pedersen, Matt Davis, Simon D. Schowanek, Scott Jarvie, Alexandre Antonelli, and Jens-Christian Svenning. 2020. <em>MegaPast2Future/PHYLACINE<span>_</span>1.2: PHYLACINE Version 1.2.1</em>. Zenodo. <a href="https://doi.org/10.5281/zenodo.3690867">https://doi.org/10.5281/zenodo.3690867</a>.
</div>
<div id="ref-hassanin2021" class="csl-entry" role="doc-biblioentry">
Hassanin, Alexandre, Géraldine Veron, Anne Ropiquet, Bettine Jansen van Vuuren, Alexis Lécu, Steven M. Goodman, Jibran Haider, and Trung Thanh Nguyen. 2021. <span>“Evolutionary History of Carnivora (Mammalia, Laurasiatheria) Inferred from Mitochondrial Genomes.”</span> <em>PLOS ONE</em> 16 (2): e0240770. <a href="https://doi.org/10.1371/journal.pone.0240770">https://doi.org/10.1371/journal.pone.0240770</a>.
</div>
<div id="ref-hirt2020" class="csl-entry" role="doc-biblioentry">
Hirt, Myriam R., Marlee Tucker, Thomas Müller, Benjamin Rosenbaum, and Ulrich Brose. 2020. <span>“Rethinking Trophic Niches: Speed and Body Mass Colimit Prey Space of Mammalian Predators.”</span> <em>Ecology and Evolution</em> 10 (14): 7094–7105. <a href="https://doi.org/10.1002/ece3.6411">https://doi.org/10.1002/ece3.6411</a>.
</div>
<div id="ref-poelen2014" class="csl-entry" role="doc-biblioentry">
Poelen, Jorrit H., James D. Simons, and Chris J. Mungall. 2014. <span>“Global Biotic Interactions: An Open Infrastructure to Share and Analyze Species-Interaction Datasets.”</span> <em>Ecological Informatics</em> 24 (November): 148–59. <a href="https://doi.org/10.1016/j.ecoinf.2014.08.005">https://doi.org/10.1016/j.ecoinf.2014.08.005</a>.
</div>
<div id="ref-soria2021" class="csl-entry" role="doc-biblioentry">
Soria, Carmen D., Michela Pacifici, Moreno Di Marco, Sarah M. Stephen, and Carlo Rondinini. 2021. <span>“COMBINE: A Coalesced Mammal Database of Intrinsic and Extrinsic Traits.”</span> <em>Ecology</em> 102 (6): e03344. <a href="https://doi.org/10.1002/ecy.3344">https://doi.org/10.1002/ecy.3344</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>